File: LeetCode/Medium/0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        char[] ch = s.toCharArray();
        String answer = String.valueOf(ch[0]);
        
        for (int i = 2; i <= ch.length; i++) {
            for (int j = 0; j <= ch.length - i; j++) {
                int left = j;
                int right = j + i - 1;
                boolean flag = true;

                while (left < right) {
                    if (ch[left] != ch[right]) {
                        flag = false;
                        break;
                    }
                    left++;
                    right--;
                }

                if (flag) {
                    answer = s.substring(j, j + i);
                    break;
                }
            }
        }
        
        return answer;
    }
}

File: LeetCode/Medium/0005-longest-palindromic-substring/0005-longest-palindromic-substring.kt
class Solution {
    fun longestPalindrome(s: String): String {
        val ch = s.toCharArray()
        var answer = ch[0].toString()

        for (i in 2 .. ch.size) {
            for (j in 0 .. ch.size - i) {
                var left = j
                var right = j + i - 1
                var flag = true

                while (left < right) {
                    if (ch[left] != ch[right]) {
                        flag = false
                        break
                    }
                    left++
                    right--
                }

                if (flag) {
                    answer = s.substring(j, j + i)
                    break
                }
            }
        }

        return answer
    }
}